src/App.js

```diff
@@ -0,0 +1,178 @@
+  const [genres, setGenres] = useState([]);
+  let [modal, SetModal] = useState(false);
+  let [rate, setRate] = useState(0);
+  let [trailer, setTrailer] = useState('');
+
+  let popularMovies = async () => {
+    let url = `https://api.themoviedb.org/3/movie/now_playing?api_key=${apikey}&language=en-US&page=${page}`;
+    //here we get an array of 20 movies
+    let fetchedData = await fetch(url);
+    let data = await fetchedData.json();
+    movieList = movieList.concat(data.results);
+
+    movieList.map(movie => {
+      movie.genres = movie.genre_ids.map(genre => genres.find(el => el.id === genre))
+    })
+    setMovies(movieList)
+  }
+  let loadMore = () => {
+    page++;
+    popularMovies()
+    setMovies(movieList);
+  };
+
+  const fetchGenres = async () => {
+    let secondUrl = `https://api.themoviedb.org/3/genre/movie/list?api_key=${apikey}&language=en-US`
+    let secondData = await fetch(secondUrl);
+    let secondResult = await secondData.json();
+    setGenres(secondResult.genres) // will change genres value
+  }
+  useEffect(() => {
+    fetchGenres();
+  }, []); // run once after mounted , mounted is different from re-rendered
+
+  useEffect(() => {
+    popularMovies();
+  }, [genres]);
+
+  //Input range slider
+
+  let searchByRate = (value) => {
+    let rounded = Math.round(value * 10) / 10
+
+    console.log("value", rounded)
+    setRate(value)
+
+    let filteredData = movieList.filter((movie) => movie.vote_average >= value);
+    setMovies(filteredData);
+  }
+
+  console.log(movies)
+  // Loading message
+  if (movies == null) {
+    return (
+      <Spinner animation="border" role="status">
+        <span className="sr-only">Loading...</span>
+      </Spinner>
+    )
+  }
+  //Navbar: Search input HI i'M IN :.))don't break my shit :D OK WILL DO you can just watch
+  let searchByKeyword = (e) => {
+    keyword = e.target.value;
+    if (keyword === '') {
+      setMovies(movieList)
+    } else {
+      setMovies(movieList.filter((movie) => movie.title.toLowerCase().includes(keyword.toLowerCase())));
+    }
+  }
+  //Navbar: Filter Movies by Genre
+  let filteredMovies = (g) => {
+    if (g === null) {
+      setMovies(movieList)
+    } else {
+      setMovies(movieList.filter((movie) => movie.genre_ids.includes(g)));
+    }
+  }
+  //Navbar: Sort by
+  let sortByPopularity = () => {
+    let sortedMovie = [...movies].sort((a, b) => b.popularity - a.popularity);
+    setMovies(sortedMovie);
+  };
+  let sortByHighestRating = () => {
+    let sortedHighestRating = [...movies].sort((a, b) => b.vote_average - a.vote_average);
+    setMovies(sortedHighestRating);
+  };
+
+  const openModal = async (movieId) => {
+    let url = `https://api.themoviedb.org/3/movie/${movieId}/videos?api_key=${apikey}&language=en-US`;
+    let response = await fetch(url);
+    let data = await response.json();
+    console.log('hehe data video:', data)
+    setTrailer(data.results[0].key)
+    SetModal(true)
```

Comment:
you can show the modal first, and let it load in. no need to wait on 108/109 before you show the modal, right?